pkg/readiness/readiness.go

```diff
@@ -60,30 +59,48 @@ func Run(bird, bird6, felix bool) {
 // checkBIRDReady checks if BIRD is ready by connecting to the BIRD
 // socket to gather all BGP peer connection status, and overall graceful
 // restart status.
-func checkBIRDReady(ipv string) error {
+func checkBIRDReady(ipv string, thresholdTime time.Duration) error {
 	// Check for unestablished peers
 	peers, err := bird.GetPeers(ipv)
 	log.Debugf("peers: %v", peers)
 	if err != nil {
 		return err
 	}
 
+	// numEstablishedPeer keeps count of number of peers with bgp state established
+	numEstablishedPeer := 0
 	s := []string{}
+
 	for _, peer := range peers {
-		if peer.BGPState != "Established" {
+		if peer.BGPState == "Established" {
+			numEstablishedPeer += 1
+		} else if peer.BGPState != "Established" {
 			s = append(s, peer.PeerIP)
 		}
 	}
-	if len(s) > 0 {
-		return fmt.Errorf("BGP not established with %+v", strings.Join(s, ","))
-	}
 
-	// Check for GR
-	gr, err := bird.GRInProgress(ipv)
+	fmt.Printf("Number of nodes with BGP peering established = %v", numEstablishedPeer)
+
+	nodenameFileStat, err := os.Stat("/var/lib/calico/nodename")
 	if err != nil {
-		return err
-	} else if gr {
-		return errors.New("graceful restart in progress")
+		return fmt.Errorf("Error in nodename file: %v", err)
+	}
+
+	if time.Since(nodenameFileStat.ModTime()) < thresholdTime {
+		if len(s) > 0 {
```

Comment:
Think this needs a comment to explain why we're doing a different check here.
```
// When we first start up, only report ready if all our peerings are established.
// This prevents rolling update from proceeding until BGP is back up.
```

Reply:
Acknowledged and added.